src/otx/algo/anomaly/openvino_model.py

```diff
@@ -173,7 +173,7 @@ def check_if_quantized(model: openvino.Model) -> bool:
             msg = "Model is already optimized by PTQ"
             raise RuntimeError(msg)
 
-        val_dataset = data_module.train_dataloader()
+        val_dataset = data_module.val_dataloader()
```

Comment:
Does that change lead to increased accuracy on val dataset?
We use train subset for quantization of other OTX models to prevent data leakage.

Reply:
At least in anomalib I am using the test dataset for checking the performance.
In OTX, these are the commands I used for testing. Below you can see the testing performance due to the change between val/test dataset. I am assuming that having some anomalous images when optimizing has a positive effect on the performance.

# `hazelnut_large`

```bash
otx train --config src/otx/recipe/anomaly_classification/stfpm.yaml --data_root ./datasets/otx/v2/anomaly/mvtec/hazelnut_large --work_dir otx-stfpm 
```

```bash
otx test --checkpoint /home/ashwin/ote/otx-stfpm/.latest/train/best_checkpoint.ckpt --config src/otx/recipe/anomaly_classification/stfpm.yaml --data_root ./datasets/otx/v2/anomaly/mvtec/hazelnut_large --work_dir otx-stfpm
```

```bash
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│        image_AUROC        │    0.9957143068313599     │
│       image_F1Score       │    0.9718309640884399     │
│      test/data_time       │    0.1736757755279541     │
│      test/iter_time       │    0.1937289535999298     │
└───────────────────────────┴───────────────────────────┘
```

```bash
otx export --work_dir otx-stfpm --checkpoint otx-stfpm/.latest/train/best_checkpoint.ckpt
```

```bash
otx optimize --work_dir otx-stfpm --checkpoint /home/ashwin/ote/otx-stfpm/20240903_073939/exported_model.xml --data_root ./datasets/otx/v2/anomaly/mvtec/hazelnut_large
```

## val_dataloader
```bash
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│        image_AUROC        │    0.9980356693267822     │
│       image_F1Score       │    0.9857142567634583     │
│      test/data_time       │   0.0028419645968824625   │
│      test/iter_time       │   0.052088286727666855    │
└───────────────────────────┴───────────────────────────┘
```
## train_dataloader

```bash
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│        image_AUROC        │            1.0            │
│       image_F1Score       │            0.0            │
│      test/data_time       │    0.00306075531989336    │
│      test/iter_time       │   0.054295849055051804    │
└───────────────────────────┴───────────────────────────┘
```

Reply:
Thanks for the explanation. That fix is ok as a workaround.